shared/js/background/classes/https-redirects.es6.js

```diff
@@ -44,17 +59,23 @@ class HttpsRedirects {
          * Redirect loop detection is different when the request is for the main frame vs
          * any other request on the page.
          *
-         * For main frames, the redirect loop could happen like this:
-         * 1. We upgrade a page to HTTPS
-         * 2. The server gives 200 but prints out some HTML that redirects the page to HTTP
-         * 3. We try to upgrade again
+         * For main frames, the redirect loop could happen as part of several distinct hits to the same URL
+         * (e.g. we saw a case where a site returned 200 and the redirected to HTTP via Javascript)
```

Comment:
The site that was previously causing this problem (http://raisingchildren.net.au/articles/safe_water_temperature.html) has fixed it since, so we have no way to test this case anymore. Good for them, not good for us.

I'd assume the redirect loop we encountered on their site would still be able to hit the same URL 7 times in 3 seconds.